gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -714,7 +711,13 @@ MacroExpander::match_n_matches (Parser<MacroInvocLexer> &parser,
   bool did_meet_lo_bound = match_amount >= lo_bound;
   bool did_meet_hi_bound = hi_bound ? match_amount <= hi_bound : true;
 
-  return did_meet_lo_bound && did_meet_hi_bound;
+  // If the end-result is valid, then we can clear the parse errors: Since
+  // repetitions are parsed eagerly, it is okay to fail in some cases
+  auto res = did_meet_lo_bound && did_meet_hi_bound;
+  if (res)
+    parser.clear_errors ();
+
```

Comment:
do we need an else case to emit errors here now?

Reply:
I'm not sure if we want to emit errors here: Since matching the repetition failed, it might simply mean that we're matching on the wrong Macro rule:
```rust
macro_rules! m {
    ($($a:literal),*) => {};
    ($($a:literal)tok*) => {};
}

m!(15 tok 14 tok 13);
```
would fail the first rule, but match the second one. And we don't want parse errors to be emitted, we simply want to silently return false and check if the next rule matches

Reply:
ah ok i see what you mean now when I read the code again.